cli/admin/src/test/java/io/pravega/cli/admin/dataRecovery/DataRecoveryTest.java

```diff
@@ -0,0 +1,490 @@
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.segmentstore.server.store.ServiceConfig;
+import io.pravega.segmentstore.server.writer.WriterConfig;
+import io.pravega.segmentstore.storage.DurableDataLogException;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperConfig;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperServiceRunner;
+import io.pravega.storage.filesystem.FileSystemStorageConfig;
+import io.pravega.storage.filesystem.FileSystemStorageFactory;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.integration.demo.ControllerWrapper;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Properties;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Tests Data recovery commands.
+ */
+@Slf4j
+public class DataRecoveryTest {
+    private static final Duration TIMEOUT = Duration.ofMillis(30 * 1000);
+    private static final int NUM_EVENTS = 10;
+    private static final String EVENT = "12345";
+    private static final String SCOPE = "testScope";
+    // Setup utility.
+    private static final Duration READ_TIMEOUT = Duration.ofMillis(1000);
+    private static final AtomicReference<AdminCommandState> STATE = new AtomicReference<>();
+
+    @Rule
+    public final Timeout globalTimeout = new Timeout(120, TimeUnit.SECONDS);
+
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(10, "Data recovery test pool");
+    private final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(1);
+    private final StreamConfiguration config = StreamConfiguration.builder().scalingPolicy(scalingPolicy).build();
+
+    /**
+     * A directory for FILESYSTEM storage as LTS.
+     */
+    private File baseDir = null;
+    private FileSystemStorageConfig adapterConfig;
+    private StorageFactory storageFactory = null;
+
+    /**
+     * A directory for storing logs and CSV files generated during the test..
+     */
+    private File logsDir = null;
+    private BookKeeperLogFactory factory = null;
+
+    @Before
+    public void setUp() throws Exception {
```

Comment:
It is a best practice to keep your tearDown next to your setUp methods. That way you can clearly see what you create and what you cleanup.